### PR TITLE
Wrap generated calls

### PR DESCRIPTION
**Pull request by vanessa-passage**
# Description

This work was over looked. All appi calls need to be wrapped. We only want users to. provide their app id on initialization. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Infrastructure

# Testing

## Checklist before approving (including SDK updates)
- [ ] I have performed a review of the code.
- [ ] Docs have been updated. Docs in the docs/Passage folder should be edited with any changes to the api. We manually edit these docs to account for the wrapping layer of the generated code.
- [ ] The version has been updated in the `composer.json` file.
- [ ] The `CHANGELOG.md` has been updated.

### CODE REVIEW

File: custom/lib/controllers/Passage.php [hunk 1]
```diff
@@ -25,5 +36,85 @@ public function __construct(string $appId, string $apiKey) {
     public function getAppId(): string {
         return $this->appId;
     }
+
+    /**
+     * Get AppId key for this Passage instance
+     * @return AppResponse Passage App
+     */
+    public function getApp() {
+        return UsersApi::getApp($this->app_id);
```
Comment (danilo-kaltner) [inline custom/lib/controllers/Passage.php:45]: Since

**Comment by danilo-kaltner** [comment]
Another thing I noticed is that there are some changes that need to be done in the Readme.md file. [Specifically in this section](https://github.com/passageidentity/passage-php/blob/PSG-2913-wrap-api-calls/README.md#getting-started)

**Review by danilo-kaltner — APPROVED** [review]

**Comment by vanessa-passage** [comment]
@danilo-kaltner please re-review. I added tests and updated the docs.

File: generated/lib/Model/EmailSmsAuthMethod.php [hunk 2]
```diff
@@ -367,9 +367,9 @@ public function setTtl($ttl)
             throw new \InvalidArgumentException('non-nullable ttl cannot be null');
         }
 
-        if (($ttl < 60)) {
-            throw new \InvalidArgumentException('invalid value for $ttl when calling EmailSmsAuthMethod., must be bigger than or equal to 60.');
-        }
+        // if (($ttl < 60)) {
```
Comment (danilo-kaltner) [inline generated/lib/Model/EmailSmsAuthMethod.php:370]: You can either remove those comments or turn those them on if they are useful

File: generated/lib/ObjectSerializer.php [hunk 3]
```diff
@@ -93,10 +93,10 @@ public static function sanitizeForSerialization($data, $type = null, $format = n
                         if (is_callable($callable)) {
                             /** array $callable */
                             $allowedEnumTypes = $callable();
-                            if (!in_array($value, $allowedEnumTypes, true)) {
-                                $imploded = implode("', '", $allowedEnumTypes);
-                                throw new \InvalidArgumentException("Invalid value for enum '$openAPIType', must be one of: '$imploded'");
-                            }
+                            // if (!in_array($value, $allowedEnumTypes, true)) {
```
Comment (danilo-kaltner) [inline generated/lib/ObjectSerializer.php:96]: Same as above

File: generated/lib/ObjectSerializer.php [hunk 4]
```diff
@@ -499,10 +499,10 @@ public static function deserialize($data, $class, $httpHeaders = null)
 
 
         if (method_exists($class, 'getAllowableEnumValues')) {
-            if (!in_array($data, $class::getAllowableEnumValues(), true)) {
-                $imploded = implode("', '", $class::getAllowableEnumValues());
-                throw new \InvalidArgumentException("Invalid value for enum '$class', must be one of: '$imploded'");
-            }
+            // if (!in_array($data, $class::getAllowableEnumValues(), true)) {
```
Comment (danilo-kaltner) [inline generated/lib/ObjectSerializer.php:502]: Same as above

**Pull request merged**